src/osd/PG.cc

```diff
@@ -1331,15 +1327,12 @@ Scrub::schedule_result_t PG::sched_scrub()
 	  << (is_clean() ? " <clean>" : " <not-clean>") << dendl;
   ceph_assert(ceph_mutex_is_locked(_lock));
 
-  if (!is_primary() || !is_active() || !is_clean()) {
-    return Scrub::schedule_result_t::bad_pg_state;
+  if (!m_scrubber || m_scrubber->is_being_scrubbed()) {
```

Comment:
Can m_scrubber actually be null here?

Reply:
PG::m_scrubber should probably have a comment explaining that it's only null during construction and destruction.

Reply:
fixed